go.mod

```
@@ -7,179 +7,176 @@ require (
 	github.com/diktyo-io/appgroup-api v1.0.1-alpha
 	github.com/diktyo-io/networktopology-api v1.0.1-alpha
 	github.com/dustin/go-humanize v1.0.1
-	github.com/go-logr/logr v1.4.1
+	github.com/go-logr/logr v1.4.2
 	github.com/google/go-cmp v0.6.0
 	github.com/k8stopologyawareschedwg/noderesourcetopology-api v0.1.2
 	github.com/k8stopologyawareschedwg/podfingerprint v0.2.2
 	github.com/patrickmn/go-cache v2.1.0+incompatible
-	github.com/paypal/load-watcher v0.2.3
+	github.com/paypal/load-watcher v0.2.4
 	github.com/spf13/pflag v1.0.5
-	github.com/stretchr/testify v1.8.4
+	github.com/stretchr/testify v1.9.0
 	gonum.org/v1/gonum v0.12.0
-	k8s.io/api v0.30.6
-	k8s.io/apimachinery v0.30.6
-	k8s.io/apiserver v0.30.6
-	k8s.io/client-go v0.30.6
-	k8s.io/code-generator v0.30.6
-	k8s.io/component-base v0.30.6
-	k8s.io/component-helpers v0.30.6
-	k8s.io/klog/hack/tools v0.0.0-20210917071902-331d2323a192
-	k8s.io/klog/v2 v2.120.1
-	k8s.io/kube-scheduler v0.30.6
-	k8s.io/kubernetes v1.30.6
-	k8s.io/utils v0.0.0-20230726121419-3b25d923346b
-	sigs.k8s.io/controller-runtime v0.18.5
+	k8s.io/api v0.31.2
+	k8s.io/apimachinery v0.31.2
+	k8s.io/apiserver v0.31.2
+	k8s.io/client-go v0.31.2
+	k8s.io/code-generator v0.31.2
+	k8s.io/component-base v0.31.2
+	k8s.io/component-helpers v0.31.2
+	k8s.io/klog/v2 v2.130.1
+	k8s.io/kube-scheduler v0.31.2
+	k8s.io/kubernetes v1.31.2
+	k8s.io/utils v0.0.0-20240711033017-18e509b52bc8
+	sigs.k8s.io/controller-runtime v0.19.1
```

```suggestion
	sigs.k8s.io/controller-runtime v0.19.2
```

I think that this can be follow-ups.

It seems that v0.19.2 has been released just yesterday: https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.19.2

---

maybe let it (v0.19.2) soak for a while, and upgrade later if needed.

---

SGTM